src/pause/140C70.c

```diff
@@ -1,17 +1,287 @@
+    mapCursorYPos += phi_f6;
+    if (mapCursorXPos < 16.0f) {
+        mapCursorXPos = 16.0f;
+    }
+    if (mapCursorYPos < 8.0f) {
+        mapCursorYPos = 8.0f;
+    }
+    if (mapCursorXPos >= 316.0f) {
+        mapCursorXPos = 315.0f;
+    }
+    if (mapCursorYPos >= 308.0f) {
+        mapCursorYPos = 307.0f;
+    }
+
+    temp_f4_2 = (s32) (mapCursorXPos + D_80270704 - D_80270098);
+    temp_f12 = mapCursorYPos + D_80270708 - D_802700A0;
+    if (D_802700A8 <= temp_f4_2) {
+        D_80270704 = D_80270704 - (temp_f4_2 - D_802700A8);
+    }
+
+    if (temp_f4_2 <= D_802700B0) {
+        D_80270704 = D_80270704 - (temp_f4_2 + D_802700B8);
+    }
+
+    temp_f12 = temp_f12;
+    if (temp_f12 >= 15.0) {
+        D_80270708 = D_80270708 - (temp_f12 - 15.0);
+    }
+
+    if (temp_f12 <= -15.0) {
+        D_80270708 =  D_80270708 - (temp_f12 + 15.0);
+    }
+
+    if (D_80270704 > 0.0f) {
+        D_80270704 = 0.0f;
+    }
+
+    if (D_80270708 > 0.0f) {
+        D_80270708 = 0.0f;
+    }
+
+    if (D_80270704 < -86.0f) {
+        D_80270704 = -86.0f;
+    }
+
+    if (D_80270708 < -210.0f) {
+        D_80270708 = -210.0f;
+    }
+
+    if ((gPauseMenuPressedButtons & B_BUTTON) != 0) {
+        sfx_play_sound(0xCA);
+        gPauseMenuCurrentTab = 0;
+        return;
+    }
+
+    gPauseMenuCurrentDescIconScript = 0;
+    if (mapCursorCurrentOption == -1) {
+        gPauseMenuCurrentDescString = 0;
+        return;
+    }
+
+    gPauseMenuCurrentDescString = (mapCursorCurrentOption * 3) + 0x1D00F3;
+    if (evt_get_variable(0, EVT_STORY_PROGRESS) >= (s32)*(D_802502B8 + (mapCursorCurrentOption * ARRAY_COUNT(D_802502B8) + 3))) {
+        gPauseMenuCurrentDescString++;
+    }
+}
+
+void pause_map_update(void) {
+    f32 temp_f0;
+    f32 temp_f4;
+    f32 temp_f4_2;
+    s32 i;
+    pause_map_spaces* phi_s1;
+    f32 phi_f20;
+    f32 phi_f22;
+
+    mapCursorCurrentOption = -1;
+    D_80270724 = 0;
+    phi_s1 = (pause_map_spaces*)&D_802502B8;
+    phi_f20 = 10000.0f;
+    phi_f22 = -1.0f;
+    
+    for (i = 0; i < EVT_SAVE_FLAG_PLACES_VISITED_TOTAL; i++) {
+        if (evt_get_variable(0, i + EVT_SAVE_FLAG_PLACES_VISITED) != 0) {
+            temp_f4 = mapCursorXPos - phi_s1->xPos;
+            temp_f0 = mapCursorYPos - phi_s1->yPos;
+            temp_f4_2 = SQ(temp_f4) + SQ(temp_f0);
+            if (temp_f4_2 < 400.0f) {
+                D_80270724++;
+            }
+
+            if ((temp_f4_2 < phi_f20) && (temp_f4_2 < 200.0f)) {
+                phi_f20 = temp_f4_2;
+                phi_f22 = i;
+            }
+        }
 
-INCLUDE_ASM(s32, "pause/140C70", pause_map_handle_input);
+        phi_s1++;
+    }
+    
+    mapCursorCurrentOption = phi_f22;
+    mapCursorCurrentOptionCopy = phi_f22;
+}
 
-INCLUDE_ASM(s32, "pause/140C70", pause_map_update);
+void pause_map_cleanup(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "pause/140C70", pause_map_cleanup);
+    for (i = 0; i <= 0; i++) {
+        free_hud_element(D_80270700[i]);
```

Comment:
ARRAY_COUNT(D_80270700)

Reply:
sorry, I mean instead of the 0 in the for loop condition. and it'd be less than, not less than or equal to